iceoryx_binding_c/include/iceoryx_binding_c/publisher.h

```diff
@@ -24,20 +24,29 @@
 /// @brief publisher handle
 typedef struct cpp2c_Publisher* iox_pub_t;
 
+/// @brief options to be set for a publisher
+typedef struct
```

Comment:
This is done because of C compilers which follow the standard very strictly to avoid having to write ``struct iox_publisher_options_t``. I would recommend a typedef for this reason, but in C++ no typedef is needed (makes no difference).

See https://stackoverflow.com/questions/612328/difference-between-struct-and-typedef-struct-in-c